framework/src/modules/interoperability/base_interoperability_internal_methods.ts

```diff
@@ -376,10 +371,149 @@ export abstract class BaseInteroperabilityInternalMethod extends BaseInternalMet
 	}
 
 	// Different in mainchain and sidechain so to be implemented in each module store separately
-	public abstract isLive(chainID: Buffer, timestamp?: number): Promise<boolean>;
-	public abstract sendInternal(sendContext: SendInternalContext): Promise<boolean>;
+	public async sendInternal(
+		context: MethodContext,
+		sendingAddress: Buffer,
+		module: string,
+		crossChainCommand: string,
+		receivingChainID: Buffer,
+		fee: bigint,
+		status: number,
+		params: Buffer,
+		timestamp?: number,
+	): Promise<void> {
+		const ownChainAccount = await this.stores.get(OwnChainAccountStore).get(context, EMPTY_BYTES);
+		const ccm = {
+			module,
+			crossChainCommand,
+			fee,
+			nonce: ownChainAccount.nonce,
+			params,
+			receivingChainID,
+			sendingChainID: ownChainAccount.chainID,
+			status,
+		};
+		// Not possible to send messages to the own chain.
+		if (receivingChainID.equals(ownChainAccount.chainID)) {
+			this.events.get(CcmSentFailedEvent).log(
+				context,
+				{
+					ccm: { ...ccm, params: EMPTY_BYTES },
+					code: CCMSentFailedCode.INVALID_RECEIVING_CHAIN,
+				},
+				true,
+			);
+			throw new Error('Sending chain cannot be the receiving chain.');
+		}
 
-	// To be implemented in base class
-	public abstract getInboxRoot(chainID: Buffer): Promise<void>;
-	public abstract getOutboxRoot(chainID: Buffer): Promise<void>;
+		// Validate ccm size.
+		try {
+			validateFormat(ccm);
+		} catch (error) {
+			this.events.get(CcmSentFailedEvent).log(
+				context,
+				{
+					ccm: { ...ccm, params: EMPTY_BYTES },
+					code: CCMSentFailedCode.INVALID_FORMAT,
+				},
+				true,
+			);
+
+			throw new Error('Invalid CCM format.');
+		}
+		// From now on, we can assume that the ccm is valid.
+
+		// receivingChainID must correspond to a live chain.
+		const isReceivingChainLive = await this.isLive(
+			context,
+			receivingChainID,
+			timestamp ?? Date.now(),
```

Comment:
It's out of scope for this issue, but do we have issue to fix send internal?

Reply:
created issue https://github.com/LiskHQ/lisk-sdk/issues/7785

Reply:
Actually yeah we can modify that but maybe another issue as this one is getting really big. But send() method is also consistent with other methods (context, ..params)